### PR TITLE
business-central export setting UI constraction

### PR DESCRIPTION
**Pull request by DhaaraniCIT**
## Summary by CodeRabbit

- **New Features**
  - Enhanced the export settings interface with new options for date selection and export types.
  - Implemented a more dynamic and user-friendly settings form with conditional fields and improved validation.

- **Improvements**
  - Updated the naming of account fields for clarity in the export settings.
  - Optimized the user interface for setting up export configurations, providing a smoother experience.

- **Bug Fixes**
  - Fixed an issue where certain export options were not correctly displayed based on the selected expense state.

- **Documentation**
  - Updated user-facing documentation to reflect the new export settings and interface changes.

### CODE REVIEW

**Comment by DhaaraniCIT** [comment]
@coderabbitai review

**Comment by DhaaraniCIT** [comment]
@coderabbitai review

**Comment by DhaaraniCIT** [comment]
Screenshot 2023-12-06 at 10 22 00 AM
Screenshot 2023-12-06 at 10 23 29 AM

File: src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.html [hunk 1]
```diff
@@ -1 +1,175 @@
...
+        </app-configuration-step-header>
+    </div>
+    <form [formGroup]="exportSettingForm">
+        <div class="tw-p-24-px">
+            <div class="tw-mb-16-px">
+                <div class="tw-rounded-lg tw-border-separator tw-border tw-bg-configuration-bg">
+                    <app-configuration-toggle-field 
+                        [form]="exportSettingForm"
+                        [label]="'Export Reimbursable Expenses'" 
+                        [subLabel]="'Enable this to export the reimbursable expenses from ' + brandingConfig.brandName + '. If not enabled, any <b>out-of-pocket</b> expenses will not be exported to Dynamic 360 Business Central CRE.'"
```
[the remark this answers is not in the document]
Comment (ashwin1111) [reply src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.html]: remove CRE

File: src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.html [hunk 2]
```diff
@@ -2,0 +1,175 @@
...
+                        [formControllerName]="'reimbursableExportGroup'"></app-configuration-select-field>
+                    </div>
+                </div>
+            </div>
+            <div>
+                <div class="tw-rounded-lg tw-border-separator tw-border tw-bg-configuration-bg">
+                    <app-configuration-toggle-field 
+                    [form]="exportSettingForm"
+                    [label]="'Export Corporate Card Expenses'" 
+                    [subLabel]="'Enable this to export Non-Reimbursable expenses from ' + brandingConfig.brandName + '. If not enabled, any <b>Corporate Credit Card</b> expenses will not be exported to Sage Intacct.'"
```
[the remark this answers is not in the document]
Comment (ashwin1111) [reply src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.html]: +1

File: src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.html [hunk 3]
```diff
@@ -2,0 +1,175 @@
...
+                        [placeholder]="'Select export state'"
+                        [formControllerName]="'cccExpenseState'"></app-configuration-select-field>
+                    </div>
+                    <div *ngIf="exportSettingForm.value.cccExportType" class="tw-mt-16-px tw-bg-white tw-border tw-border-solid tw-border-separator tw-rounded-12-px">
+                        <app-configuration-select-field
+                        [form]="exportSettingForm"
+                        [isFieldMandatory]="true"
+                        [mandatoryErrorListName]="'export date'"
+                        [label]="'Set the ' + helper.getExportType(exportSettingForm.value.cccExportType) + ' date as'"
+                        [subLabel]="'Expenses will be grouped and posted using the configured date when exporting from Fyle to Sage Intacct. Click here for more details.'"
```
[the remark this answers is not in the document]
Comment (ashwin1111) [reply src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.html]: +1, also intacct keyword is present

File: src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.ts [hunk 4]
```diff
@@ -57,23 +58,55 @@ export class BusinessCentralExportSettingsComponent implements OnInit {
...
+  entityName: BusinessCentralExportSettingFormOption[] = [
+    {
+      label: 'Employee',
+      value: FyleField.EMPLOYEE
+    },
+    {
+      label: 'Vendor',
+      value: FyleField.VENDOR
+    }
+  ];
```
Comment (ashwin1111) [inline src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.ts:78]: get from model

File: src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.ts [hunk 5]
```diff
@@ -57,23 +58,55 @@ export class BusinessCentralExportSettingsComponent implements OnInit {
...
+
+  employeeMapOptions: BusinessCentralExportSettingFormOption[] = [
```
Comment (ashwin1111) [inline src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.ts:80]: same as above

File: src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.ts [hunk 6]
```diff
@@ -57,23 +58,55 @@ export class BusinessCentralExportSettingsComponent implements OnInit {
...
-  refreshDimensions(isRefresh: boolean) {
+  save() {
+    // TODO
+  }
+
+  getExportDate(options: BusinessCentralExportSettingFormOption[]): BusinessCentralExportSettingFormOption[]{
+    if (this.exportSettingForm.value.reimbursableExpenseState === ExpenseGroupedBy.REPORT) {
+      return options.slice(0, -1);
+    }
+    return options;
```
Comment (ashwin1111) [inline src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.ts:106]: fo .filter(). If the toggle kept on changing, we mighe be in a bad state.

**Pull request merged**